stablehlo/dialect/StablehloOps.cpp

```diff
@@ -1910,6 +1910,29 @@ LogicalResult SetDimensionSizeOp::inferReturnTypeComponents(
       adaptor.getSize(), adaptor.getDimension(), inferredReturnShapes);
 }
 
+mlir::Speculation::Speculatability SetDimensionSizeOp::getSpeculatability() {
+  // If the dimension being set is constant, the verifier will have checked that
+  // it matches the corresponding dimension in the output.
+  if (matchPattern(getSize(), m_Constant()))
+    return mlir::Speculation::Speculatable;
```

Comment:
Don't we have to do the check for remaining dimension?

Reply:
Good point! Added the missing test case and fixed. Thanks!